src/database/testInfo.ts

```diff
@@ -0,0 +1,48 @@
+/** @ignore */
+/*
+ * Copyright (c) 2025 Certinia Inc. All rights reserved.
+ */
+
+import { TestInfo } from './entity/testInfo';
+import { getConnection } from './connection';
+
+export async function saveTestInfoRecords(testStepResults: TestInfo[]) {
+  const connection = await getConnection();
+  return connection.manager.save(testStepResults);
+}
+
+export async function getTestInfoRecordThatAlreadyExist(
+  flowActionPairs: { flowName: string; actionName: string }[]
+) {
+  const connection = await getConnection();
+
+  // Generate the query for all flow-action pairs
+  const flowActionConditions = flowActionPairs
+    .map(pair => `('${pair.flowName}', '${pair.actionName}')`)
+    .join(', ');
+
+  const query = `
+    SELECT
+      id,
+      flow_name,
+      action
+    FROM performance.test_info
+    WHERE (flow_name, action) IN (${flowActionConditions})
+  `;
+
+  try {
+    const result = await connection.query(query);
+
+    const resultsMap = new Map<string, number>();
+    // Populate the results map
+    result.forEach((row: { id: number; flow_name: string; action: string }) => {
+      const key = `${row.flow_name}_${row.action}`;
+      resultsMap.set(key, row.id);
+    });
+
+    return resultsMap;
```

Comment:
This is not a consistent return with the `catch`, you may want a `Record<string, number>` instead of a `Map`. Which is `{ [key: string]: number }` like the alerts version of this file. I would qualify the return type on the function too.

e.g. usage
```ts
const resultsMap: Record<string, number> = {};
resultsMap[key] = row.id;

...
// can probably get away with undefined values
// it will still insert or update
testInfo.id = resultsMap[key];

// alternate
const id = resultsMap[key];
if (id != null) testInfo.id = id;
```